### PR TITLE
Various changes to build pipeline

### PR DESCRIPTION
**Pull request by Tirokk**
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-10T22:25:31Z" title="Wednesday, September 11th 2019, 12:25:31 am +02:00">Sep 11, 2019</time>_
_Merged <time datetime="2019-09-12T21:39:46Z" title="Thursday, September 12th 2019, 11:39:46 pm +02:00">Sep 12, 2019</time>_
---

* Use `production` docker images for cypress tests. As one of our contributors
pointed out: It is bad practice to end-to-end test sth. else than the actual
production images.
* Have a dedicated `docker-compose.build-and-test.yml` to test
backend integration tests. The backend *needs* a database when doing
integration tests. So there is no way around using docker-compose. But
we could use `production` stage when we do cypress tests.
* Remove a couple of unused ports and DRY `docker-compose.override.yml`
* Build and tag all images first and then run `docker-compose .. up` - I
hope this will avoid rebuilding the image.
* Reduce docker image size

### Issues

- Refactoring

### Todo

- [X] None

### CODE REVIEW

File: cypress.env.template.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "SEED_SERVER_HOST": "http://localhost:4001",
+  "BACKEND_HOST": "http://localhost:4000",
```
Comment (Tirokk) [inline cypress.env.template.json:2]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-12T11:00:57Z" title="Thursday, September 12th 2019, 1:00:57 pm +02:00">Sep 12, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,6 +1,6 @@
 {
-  "SEED_SERVER_HOST": "http://localhost:4001",
+  "BACKEND_HOST": "http://localhost:4000",
```


Everybody make a `cp cypress.env.template.json cypress.env.json`!

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-12T11:03:00Z" title="Thursday, September 12th 2019, 1:03:00 pm +02:00">Sep 12, 2019</time>_

---


For reference:

Screenshot - 2019-09-12T130417 098


Screenshot - 2019-09-12T130357 090

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-12T23:24:28Z" title="Friday, September 13th 2019, 1:24:28 am +02:00">Sep 13, 2019</time>_

---


Screenshot - 2019-09-13T012456 622

Screenshot - 2019-09-13T012333 293


I don't know how this is possible but `nitro-backend` stayed the same and `nitro-web` is even bigger than before!?

**Review by Mogge — CHANGES_REQUESTED** [review]
I want to test this out locally, but it's not working for me
I have copied the cypress json with `cp  cypress.env.template.json cypress.env.json`, run `docker-compose up --build`, I'm getting this error
Screenshot from 2019-09-12 19-34-35

I tried removing the directory `.nuxt` with `rm -rf webapp/.nuxt` and running `docker-compose up` with no change. 
I removed the volumes with `docker-compose down -v` which gave this feedback
```
Removing human-connection_webapp_1      ... done
Removing human-connection_backend_1     ... done
Removing human-connection_neo4j_1       ... done
Removing human-connection_maintenance_1 ... done
Removing human-connection_mailserver_1  ... done
Removing network human-connection_hc-network
Removing volume human-connection_webapp_node_modules
Removing volume human-connection_webapp_nuxt_folder
Removing volume human-connection_backend_node_modules
Removing volume human-connection_neo4j_data
Removing volume human-connection_uploads
ERROR: remove human-connection_uploads: volume is in use - [a109593c152cb5ae10fed4bdccb424212acdd88ba6c624e9966b25c7d959cd73]
```
and then `docker-compose up --build`, still no luck
something I'm doing wrong?

**Review by Mogge — APPROVED** [review]
looks good to me @roschaefer 
tested docker locally and it works as expected, and it's passing the build, so that's a good sign

**Pull request closed without merging**